front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -36,22 +36,39 @@ import type {
   LayerRangeData,
   WaypointsPanelData,
 } from 'modules/simulationResult/types';
-import type { TimetableItemId, TrainId, TrainScheduleId } from 'reducers/osrdconf/types';
+import computeOccurrenceName from 'modules/trainschedule/helpers/computeOccurrenceName';
+import type {
+  OccurrenceId,
+  TimetableItemId,
+  TrainId,
+  TrainScheduleId,
+} from 'reducers/osrdconf/types';
 import { updateSelectedTrainId } from 'reducers/simulationResults';
 import { useAppDispatch } from 'store';
-import { formatTrainScheduleIdToEditoastTrainId } from 'utils/trainId';
+import { Duration } from 'utils/duration';
+import { getPacedTrainIdFromOccurrenceId, isOccurrence } from 'utils/trainId';
 
 import SettingsPanel from './SettingsPanel';
-import { getIdFromTrainPath, getPathStyle } from './utils';
+import getPathStyle from './utils';
 import ManchetteMenuButton from '../SpaceTimeChart/ManchetteMenuButton';
 import ProjectionLoadingMessage from '../SpaceTimeChart/ProjectionLoadingMessage';
 import useWaypointMenu from '../SpaceTimeChart/useWaypointMenu';
 import WaypointsPanel from '../SpaceTimeChart/WaypointsPanel';
 
+export type TrainSpaceTimeDataWithPaced = TrainSpaceTimeData & {
```

Comment:
Why not add directly these new props to `TrainSpaceTimeData` ?
```
export type TrainSpaceTimeData = {
  name: string;
  spaceTimeCurves: {
    positions: number[];
    times: number[];
  }[];
  departureTime: Date;
  signalUpdates: ProjectPathTrainResult['signal_updates'];
} & (
  | { id: TrainScheduleId }
  | {
      id: OccurrenceId;
      paced: {
        duration: string;
        step: string;
      };
    }
);
```

Reply:
sure, done